boards/arm/olimex_stm32_p103/doc/index.rst

```diff
@@ -0,0 +1,267 @@
+.. _olimex_stm32_p103:
+
+OLIMEX-STM32-P103
+#################
+
+Overview
+********
+
+The OLIMEX-STM32-P103 board is based on the STMicroelectronics STM32F103RB ARM
+Cortex-M3 CPU.
+
+.. figure:: img/olimex_stm32_p103.png
+     :width: 540px
+     :align: center
+     :height: 454px
+     :alt: OLIMEX-STM32-P103
+
+     OLIMEX-STM32-P103
+
+Hardware
+********
+
+Information about the board can be found at the
+`OLIMEX-STM32-P103 website`_ and `OLIMEX-STM32-P103 user manual`_.
+The `ST STM32F103RB Datasheet`_ contains the processor's
+information and the datasheet.
+
+Supported Features
+==================
+
+The olimex_stm32_p103 board configuration supports the following
+hardware features:
+
++-----------+------------+----------------------+
+| Interface | Controller | Driver/Component     |
++===========+============+======================+
+| NVIC      | on-chip    | nested vectored      |
+|           |            | interrupt controller |
++-----------+------------+----------------------+
+| SYSTICK   | on-chip    | system clock         |
++-----------+------------+----------------------+
+| UART      | on-chip    | serial port          |
++-----------+------------+----------------------+
+| GPIO      | on-chip    | gpio                 |
++-----------+------------+----------------------+
+
+Other hardware features are not supported by the Zephyr kernel.
```

Comment:
That's not entirely true, as Zephyr should support quite a lot of this MCU. The features are just not configured in the devicetree of this board.